contrib_bots/lib/CommuteBot/CommuteBot.py

```diff
@@ -0,0 +1,215 @@
+        place of the space/s.
+        A departure time and a arrival time can not be inputted at the same time
+        To add more than 1 input for a category,
+        e.g. more than 1 destinations,
+        use (|), e.g. destinations=Trump+Tower|The+White+House
+        No spaces within addresses.
+        Departure times and arrival times must be in the UTC timezone,
+        you can use the timezone bot.
+                            '''
+            self.send_info(message, help_info, client)
+        else:
+            params = {}
+            # converts input parameters for api into a dictionary
+            # structure for more aethestic coding
+            for item in content_list:
+                # enables key value pair
+                org = item.split('=')
+                # ensures that invalid inputs are not entered into url request
+                if len(org) != 2:
+                    continue
+                key, value = org
+                params.update({key: value})
+
+            # limited to UTC timezone because of difficulty with user inputting
+            # correct string for input
+            if 'departure_time' in params:
+                if params['departure_time'] != 'now':
+                    departure_time = params['departure_time']
+                    times = departure_time.split(',')
+                    times = [int(x) for x in times]
+                    a = dt.datetime(1970, 1, 1, 0, 0, 0)
+                    b = dt.datetime(*times)
+                    total_seconds = str(int((b-a).total_seconds()))
+                    params.update({'departure_time': total_seconds})
+
+            if 'arrival_time' in params:
+                arrival = params['arrival_time']
+                times = arrival.split(',')
+                times = [int(x) for x in times]
+                a = dt.datetime(1970, 1, 1, 0, 0, 0)
+                b = dt.datetime(*times)
+                total_seconds = str(int((b-a).total_seconds()))
+                params.update({'arrival_time': total_seconds})
+
+            # adds API Authentication Key to url request
+            config = cp.ConfigParser()
+            home = expanduser('~')
+            settings = open(home + '/settings.ini')
+            config.readfp(settings)
+            api_key = config.get('Google.com', 'api_key')
+            params.update({'key': api_key})
+            settings.close()
+
+            # sends url api request and converts response into json format
+            r = requests.get('https://maps.googleapis.com/maps/api/'
+                             + 'distancematrix/json',
+                             params=params)
+            if r.status_code == 200:
+                rjson = r.json()
+            else:
+                self.send_info(message,
+                               "Something went wrong. Please try again."
+                               + " Error: {error_num}."
+                               .format(error_num=r.status_code), client)
```

Comment:
You could have the `r.text`, which is the response body, as well for allowing better troubleshooting for the user.